src/column/reader.rs

```diff
@@ -164,10 +164,25 @@ impl<'a, T: DataType> ColumnReaderImpl<'a, T> where T: 'static {
       }
 
       // Batch size for the current iteration
-      let iter_batch_size = ::std::cmp::min(
-        batch_size,
-        (self.num_buffered_values - self.num_decoded_values) as usize
-      );
+      let iter_batch_size = {
+        // Compute approximate value based on values decoded so far
+        let mut adjusted_size = ::std::cmp::min(
```

Comment:
nit: we have quite a lot `::std::cmp::min` and similar now - perhaps we can import it can replace them with `min` - makes it easier for reading.

Reply:
Done.